src/wallet.js

```diff
@@ -140,27 +140,37 @@ const wallet = {
    * Start a new HD wallet from an xpriv.
    * Encrypt this private key and save data in storage
    *
-   * @param {string} xpriv_root Root of the extended private-key
+   * @param {string} xpriv Extended private-key to start wallet
    * @param {string} pin
    * @param {boolean} loadHistory if should load the history from the generated addresses
    *
    * @return {Promise} Promise that resolves when finishes loading address history, in case loadHistory = true, else returns null
    * @memberof Wallet
    * @inner
    */
-  executeGenerateWalletFromXPriv(xpriv_root, pin, loadHistory) {
-    let xpriv = HDPrivateKey(xpriv_root);
-    let privkey = xpriv.derive(`m/44'/${HATHOR_BIP44_CODE}'/0'/0`);
+  executeGenerateWalletFromXPriv(xprivkey, pin, loadHistory) {
+    const xpriv = HDPrivateKey(xprivkey);
+    let initialAccessData;
+    let privkey;
+    if (xpriv.depth === 0) {
+      // TODO test if it's working for root xpriv
```

Comment:
TODO?!

Reply:
This already had a unit test. Removed the TODO